### PR TITLE
Fix HttpServletMapping.getPattern for /* mapping

### PR DESCRIPTION
**Pull request by pmd1nh**
https://github.com/OpenLiberty/open-liberty/issues/8085

### CODE REVIEW

**Comment by pmd1nh** [comment]
#build        #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.1,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

File: dev/com.ibm.ws.webcontainer.servlet.4.0/src/com/ibm/ws/webcontainer40/srt/SRTServletRequest40.java [hunk 1]
```diff
@@ -186,8 +186,10 @@ public HttpServletMapping getCurrentHttpServletMapping(WebAppDispatcherContext40
                     returnMapping = new HttpServletMappingImpl(MappingMatch.EXTENSION, matchValue, pattern, servletName);
                     break;
                 case PATH:
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.4.0/src/com/ibm/ws/webcontainer40/srt/SRTServletRequest40.java]: Hey Phu can you just add a comment here so we know in the future why we're only changing the matchValue when there isn't a slashStarMapping?

File: dev/com.ibm.ws.webcontainer.servlet.4.0/src/com/ibm/ws/webcontainer40/srt/SRTServletRequest40.java [hunk 2]
```diff
@@ -186,8 +186,10 @@ public HttpServletMapping getCurrentHttpServletMapping(WebAppDispatcherContext40
...
-                    // matchValue is the pathInfo after the last "/" and pattern is the servletPath + "/*"
-                    matchValue = pathInfo.substring(pathInfo.lastIndexOf("/") + 1, pathInfo.length());
+                    if (!dispatchContext.hasSlashStarMapping()) {
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.4.0/src/com/ibm/ws/webcontainer40/srt/SRTServletRequest40.java]: Hey Phu can you just add a comment here so we know in the future why we're only changing the matchValue when there isn't a slashStarMapping?

File: dev/com.ibm.ws.webcontainer.servlet.4.0/src/com/ibm/wsspi/webcontainer40/util/URIMatcher40.java [hunk 3]
```diff
@@ -203,13 +203,21 @@ public Object match(IExtendedRequest req) {
...
         // hit the defaultNode "/*"
         if (defaultNode != null) {
-
             // Servlet 4.0: default match
-            dispatchContext.setMappingMatch(MappingMatch.DEFAULT);
-            if (TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
-                logger.logp(Level.FINE, CLASS_NAME, methodName, "set MappingMatch to: " + MappingMatch.DEFAULT);
-            }
+            // Two possible here: 1. /* mapping.  This can be considered as a special MappingMatch.PATH with the PATH happens to be just the root /
+            //                    2. / (default servlet) mapping
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.4.0/src/com/ibm/wsspi/webcontainer40/util/URIMatcher40.java]: Change a bit:

possible -> possibilities
as a -> a
with -> where

**Comment by pmd1nh** [comment]
All tests passed.

**Comment by pmd1nh** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.1,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by pmd1nh** [comment]
one failed test in:
	
com.ibm.ws.microprofile.reactive.messaging_fat FAT tests
junit.framework.AssertionFailedError: Timeout occurred. Please note the time in the report does not reflect the time until the timeout.
	at jdk.internal.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at com.ibm.aries.buildtasks.Try.execute(Try.java:119)

It is not related to this PR.

All other tests (included new test) passed.

File: dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCGetMappingSlashStarTest.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCGetMappingSlashStarTest.java]: This is a new class so there should not be a date range in the copyright, or am I missing something?

**Comment by pmd1nh** [comment]
POC approved.  CTS tests passed.

**Pull request merged**